client/state/plugins/marketplace/reducer.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import type { AnyAction } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { PLUGINS_MARKETPLACE_PRIMARY_DOMAIN_UPDATE } from 'calypso/state/action-types';
+import { combineReducers } from 'calypso/state/utils';
+
+export interface IPurchaseFlowState {
+	primaryDomain: string | undefined;
+}
+
+const defaultState: IPurchaseFlowState = {
+	primaryDomain: undefined,
+};
+
+function purchaseFlow(
+	state: IPurchaseFlowState = defaultState,
+	action: AnyAction
+): IPurchaseFlowState {
+	switch ( action.type ) {
+		case PLUGINS_MARKETPLACE_PRIMARY_DOMAIN_UPDATE:
+			return {
+				...state,
```

Comment:
I assume this is pre-empting a larger reducer that is going to evolve in future PRs? If we're not sure that it will grow over time, keeping it simple is preferred.